app/HTTPServer.js

```diff
@@ -1,42 +1,107 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-const constants = require('./constants')
-const http = require('http')
+const express = require('express')
+const IoTHubTokenCredentials = require('azure-iothub').IoTHubTokenCredentials
+const DigitalTwinServiceClient = require('azure-iothub').DigitalTwinClient
+const { parseConnectionString } = require('rhea-promise')
+
 const { inspect } = require('util')
 const url = require('url')
-const PORT = 8080
+const path = require('path')
+const iotHubConfiguration = require('./iotHubConfiguration')
+const influxwriter = require('./influxReaderWriter')
+
 const TYPE_PROPERTY = 'Property'
 const TYPE_COMMAND = 'Command'
 const ROOT_COMPONENT = 'Root'
 
-const propertiesCommandsAPI = function (dtServiceclient)
+async function expressAppConfig ()
 {
-    http.createServer(function (req, res)
-    {
-        res.writeHead(200, {})
-        res.end()
+    const app = express()
+    app.use(express.urlencoded())
 
-        console.log('Received URL: %s', req.url)
-        const query = url.parse(req.url, true).query
+    app.get('/configuration-form', function (req, res)
+    {
+        res.sendFile(path.join(__dirname, './form.html'))
+    })
 
-        console.log(query.type)
-        console.log(query.componentName)
-        console.log(query.methodName)
-        console.log(query.value)
+    app.post('/submit-form', function (req, res)
+    {
+        let body = ''
+        console.log('Form Submitted!')
+        console.log('Connection String: ', req.body.connectionstring)
+        console.log('Device ID: ', req.body.deviceid)
 
-        if (query.type === TYPE_PROPERTY)
+        const { HostName, SharedAccessKeyName, SharedAccessKey } = parseConnectionString(
+            req.body.connectionstring
+        )
+        // Verify that the required info is in the connection string.
+        if (!HostName || !SharedAccessKey || !SharedAccessKeyName)
         {
-            UpdateDigitalTwin(dtServiceclient, query.componentName, query.methodName, query.value)
-            console.log('Updating Property..')
+            body = 'Invalid IoT Hub Connection String! <a href="http://localhost:8080/configuration-form">Reopen Solution Sample Configuration</a>'
```

Comment:
Update this to http://localhost:8080 if the above change request makes sense.

Reply:
I think this caused a problem because the commands sent from the Grafana are sent to this top level URL.

Reply:
I reverted this back because it was blocking my testing.

Reply:
Here is where this URL is used: https://github.com/Azure/Verified-Telemetry-Solution-Sample/blob/9c62cb0f48249cfc8470a660319ff7e3e2518895/grafana/provisioning/dashboards/VerifiedTelemetryGSGDashboard.json#L206